modules/core/src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -554,6 +547,10 @@ object KafkaConsumer {
 
       override def toString: String =
         "KafkaConsumer$" + id
+
+      override def terminate: F[Unit] = fiber.cancel
+
+      override def awaitTermination: F[Unit] = fiber.join
```

Comment:
By "inlining the combine operation" do you mean the following @bplommer?
```suggestion
      override def terminate: F[Unit] = (actor combine polls).cancel

      override def awaitTermination: F[Unit] = (actor combine polls).join
```

Reply:
No, I meant inline the code from `cats-effect` and `cats` that implements `combine` on fibers. So I think that would be code from the `Applicative` instance for `Fiber` and the `Semigroup` instance for `Applicative`. Then `KafkaConsumer#fiber` could just be defined as `Fiber(terminate.start.void, awaitTermination)` and would be easy to completely remove for v2.

Reply:
Ok, sorry I'm not getting it @bplommer  😕 I'm assuming that `KafkaConsumer` will still be passed `polls: Fiber[F, Unit]` and `actor: Fiber[F, Unit]`? If that's the case then isn't it best to just use the `combine` functionality rather than inline it?

Will change the definition of `fiber = Fiber(terminate.start.void, awaitTermination)`

Reply:
Actually, should `val fiber = fiber = Fiber(awaitTermination.start.void, terminate)` instead, if `terminate` is cancellation and `awaitTermination` is joining?

Reply:
Yes, they'll still be passed `polls` and `actor` - these are "real" fibers produced by the CE runtime system. But `pollsFiber` and `actorFiber` are "fake" fibers, in the sense that they're created by synthesising `join` and `cancel` operations rather than by invoking `.start` on an effect. 

This (a) is kind of hacky and (b) doesn't work in CE3, where you can't synthesise fibers in the same way. (b) means that if we want the same behaviour in CE3 we'll need to implement it ourselves anyway - so for consistency across branches it's better to do it in 1.x.

Reply:
> Actually, should `val fiber = Fiber(awaitTermination.start.void, terminate)` instead, if `terminate` is cancellation and `awaitTermination` is joining?

Sorry, yes - I got the order wrong there.

Reply:
Thanks @bplommer, I've made some updates: https://github.com/fd4s/fs2-kafka/pull/504/commits/dbf05abbfd2c5fb1c8f48e629ca32b7430b2d9cc
but I didn't understand the inlining work to do. I think you're suggesting we add logic similar to https://github.com/typelevel/cats-effect/blob/series/2.x/core/shared/src/main/scala/cats/effect/Fiber.scala#L109-L117
To me I'm not sure why we don't just use the existing `combine` method.

Reply:
Yes, that's the bit of logic I'm talking about inlining.

To be honest, for the CE2-based versions of fs2-kafka I think this is an implementation detail that doesn't hugely matter. It's probably best if we keep the existing `combine` for now and then think about backporting whatever changes we make in the CE3 branch.

Reply:
Great! Thanks @bplommer.